frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/api/KafkaZKClient.java

```diff
@@ -28,61 +28,72 @@
 
     private final CuratorFramework zkClient;
     private final String kafkaZkUri;
+    private final String kafkaServicePath;
 
-    public KafkaZKClient(String kafkaZkUri) {
+    public  String getZKString() {
+        return  kafkaZkUri + kafkaServicePath;
+    }
+
+    public KafkaZKClient(String kafkaZkUri, String servicePath) {
         if (kafkaZkUri == null){
-            throw new InvalidParameterException("Kafka Zookeeper path is not set. Can not start client.");
+            throw new InvalidParameterException("Zookeeper URI is not set. Can not start client.");
+        }
+        if (servicePath == null){
+            throw new InvalidParameterException("Service Path  is not set. Can not start client.");
         }
         this.zkClient = CuratorFrameworkFactory.newClient(
                 kafkaZkUri,
                 new ExponentialBackoffRetry(POLL_DELAY_MS, CURATOR_MAX_RETRIES));
         this.zkClient.start();
         this.kafkaZkUri = kafkaZkUri;
-    }
-
-    public String getZKString(){
-        return kafkaZkUri;
+        this.kafkaServicePath = servicePath;
+        log.info("Zookeeper Client URI: " + kafkaZkUri + " servicePath : " + kafkaServicePath);
     }
 
     public JSONArray listBrokers() throws Exception {
         try {
-            return new JSONArray(zkClient.getChildren().forPath(kafkaZkUri + BROKER_ID_PATH));
+            log.info("Zookeeper Client list brokers path: " + kafkaServicePath + BROKER_ID_PATH);
+            return new JSONArray(zkClient.getChildren().forPath(kafkaServicePath + BROKER_ID_PATH));
         } catch (KeeperException.NoNodeException e) {
-            log.info("List path: " + kafkaZkUri + BROKER_ID_PATH
+            log.info("List path: " + kafkaServicePath + BROKER_ID_PATH
```

Comment:
Can format as `log.info(String.format("List path: %s%s doesn't exist, returning empty brokers list. Kafka not running yet?", kafkaServicePath, BROKER_ID_PATH), e);`